interp/declare.ml

```diff
@@ -382,17 +382,36 @@ let inInductive : inductive_obj -> obj =
     discharge_function = discharge_inductive;
     rebuild_function = infer_inductive_subtyping }
 
-let declare_projections mind =
-  let spec,_ = Inductive.lookup_mind_specif (Global.env ()) (mind,0) in
+let declare_projections univs mind =
+  let env = Global.env () in
+  let spec,_ = Inductive.lookup_mind_specif env (mind,0) in
     match spec.mind_record with
-    | Some (Some (_, kns, pjs)) ->
-      Array.iteri (fun i kn ->
+    | Some (Some (_, kns, pbs)) ->
+      let projs = Inductiveops.compute_projections env (mind, 0) in
+      let i = ref 0 in
+      Array.iter2 (fun kn pb ->
```

Comment:
Why use a ref instead of iteri or iter3?

Reply:
There are no such combinators...

Reply:
So add them.

Reply:
The code can be simplified though.

Reply:
>The code can be simplified though.

For instance by not having some random ref.